docs/05-infrastructure-nodes.md

```diff
@@ -0,0 +1,318 @@
+        sigs.k8s.io/cluster-api-machineset: infrastructure-us-west-1b
+    spec:
+      metadata:
+        creationTimestamp: null
+        labels:
+          infra: "true"
+      providerSpec:
+        value:
+          ami:
+            id: ami-08871aee06d13e584
+          apiVersion: awsproviderconfig.k8s.io/v1alpha1
+          blockDevices:
+          - ebs:
+              iops: 0
+              volumeSize: 32
+              volumeType: gp2
+          deviceIndex: 0
+          iamInstanceProfile:
+            id: 190125-3-worker-profile
+          instanceType: m4.large
+          kind: AWSMachineProviderConfig
+          metadata:
+            creationTimestamp: null
+          placement:
+            availabilityZone: us-west-1b
+            region: us-west-1
+          publicIp: null
+          securityGroups:
+          - filters:
+            - name: tag:Name
+              values:
+              - 190125-3_worker_sg
+          subnet:
+            filters:
+            - name: tag:Name
+              values:
+              - 190125-3-worker-us-west-1b
+          tags:
+          - name: openshiftClusterID
+            value: 45d08e94-6bf6-4fd3-988f-54a616d04252
+          - name: kubernetes.io/cluster/190125-3
+            value: owned
+          userDataSecret:
+            name: worker-user-data
+      versions:
+        kubelet: ""
+```
+
+## Create Your Machineset
+Now you can create your `MachineSet`:
+
+```bash
+oc create -f infra-machineset.yaml
+```
+
+Then go ahead and `oc get machineset -n openshift-cluster-api` and you should
+see it listed:
+
+```
+NAME                         DESIRED   CURRENT   READY     AVAILABLE   AGE
+190125-3-worker-us-west-1b   2         2         2         2           4h
+190125-3-worker-us-west-1c   1         1         1         1           4h
+infrastructure-us-west-1b    1         1                               4s
+```
+
+We don't yet have any ready or available machines in the set because the
+instance is still coming up and bootstrapping. You can check `oc get machine
+-n openshift-cluster-api` to see when the instance finally starts running.
+Then, you can use `oc get node` to see when the actual node is joined and
+ready. If you're having trouble figuring out which node is the new one, take
+a look at the `AGE` column. It will be the youngest!
+
+## Check the Labels
+In our case, the youngest node was named `ip-10-0-128-138.us-west-1.compute.internal`, so we can ask what its labels are:
+
+    oc get node ip-10-0-128-138.us-west-1.compute.internal --show-labels
+
+And, in the `LABELS` column we see:
+
+    beta.kubernetes.io/arch=amd64,beta.kubernetes.io/instance-type=m4.large,beta.kubernetes.io/os=linux,failure-domain.beta.kubernetes.io/region=us-west-1,failure-domain.beta.kubernetes.io/zone=us-west-1b,infra=true,kubernetes.io/hostname=ip-10-0-128-138,node-role.kubernetes.io/worker=
+
+It's hard to see, but our `infra=true` label is there. Success!
```

Comment:
Maybe `acme.example.com/role=infra` to get sorted at the front of the list? ;)  A generic role-selection widget is probably better than an `infra` boolean anyway, and it gets you out of having to talk about YAML strings vs. booleans.  You may be able to recycle `node-role.kubernetes.io` as well, but I'm not sure if that has any side-effects I'm not aware of.

Reply:
I was going to suggest setting `node-role.kubernetes.io/infra=""` as that makes it show up nicely in `oc get node`

Reply:
@eparis when i was digging into the machineconfig stuff and thinking forwards towards `sysctl` with MCO I had a concern about how roles are mapped with configs and what would happen if I changed the actual role from "worker" to "infra"?

Reply:
> And the node list looks like:
>
>      ip-10-0-217-226.ec2.internal   NotReady   infra,worker   14s       v1.11.0+406fc897d8

An implicit worker role sounds like a bug to me.  What if you wanted to boot machines that don't end up becoming Kubernetes nodes?  I'd be cautious about committing to this behavior in docs.